pkg/controller/nodes/task/taskexec_context.go

```diff
@@ -123,7 +123,8 @@ func (t *Handler) newTaskExecutionContext(ctx context.Context, nCtx handler.Node
 		return nil, err
 	}
 
-	ow := ioutils.NewBufferedOutputWriter(ctx, ioutils.NewRemoteFileOutputPaths(ctx, nCtx.DataStore(), nCtx.NodeStatus().GetOutputDir()))
+	// Append backslash to the outputDir
+	ow := ioutils.NewBufferedOutputWriter(ctx, ioutils.NewRemoteFileOutputPaths(ctx, nCtx.DataStore(), nCtx.NodeStatus().GetOutputDir()+"/"))
```

Comment:
I would prefer the fix being applied to where the output dir was generated originally.

Reply:
Yes. But the would require some changes to `ConstructReference` which will require some changes in the flytestdlib.

Reply:
Then let's do it. :) Fixing here feels like a bandaid really.

Reply:
:disappointed: I don't have enough time to finish this before my vacation. Can you take over this?

Reply:
We can move forward with lyft/flytekit#84 first and I can rework on this PR.